### ISSUE TITLE
Retrieving Heat Stack outputs

### ISSUE THREAD

**Issue by piyush82** [ContainX/openstack4j#378]
I can create and delete Heat Stacks from a static Yaml file, but no matter what I do, I can not read the output list, it always returns null. Please advise.

**Comment by piyush82** [comment]
HeatStackCreate.HeatStackCreateConcreteBuilder hBuilder = new HeatStackCreate.HeatStackCreateConcreteBuilder();
                hBuilder.templateFromFile(testHOTPath);
                hBuilder.timeoutMins(5L);
                hBuilder.name("test-stack");
                StackCreate sc = hBuilder.build();
                HeatStack stack = (HeatStack) os.heat().stacks().create(sc);
                //System.exit(0);

                //Stack stack = os.heat().stacks().create(Builders.stack().name("test-stack").template(hotContent).timeoutMins(5L).build());
                System.out.println("Sent the template for creation, id: " + stack.getId());
                Thread.sleep(5000);

Until this point things work fine, I get the newly created stack ID, but then when i do:
stack.getOutputs(), I always get a null response.

**Comment by piyush82** [comment]
I do have a valid HOT with an output specified: 
heat_template_version: 2013-05-23

description: It is just a simply test HOT

resources:
  my_instance:
    type: OS::Nova::Server
    properties:
      key_name: my-key
      image: some-image-id
      flavor: m1.small
      networks: [{"network": "some-network-id"}]

outputs:
  instance_ip:
    description: The IP address of the deployed instance
    value: { get_attr: [my_instance, first_address] }

**Comment by gondor** [comment]
I think your trying too soon and the Stack hasn't been fully created.  You will usually get null outputs until the stack status is CREATE_COMPLETE.  I tested this and and waited for the CREATE_COMPLETE status and the outputs were all there.

Let me know

**Comment by vinodborole** [comment]
@piyush82 closing this issue for now, please reopen it if you feel its still an issue

**Comment by Suavac** [comment]
I've just encountered a similar issue. After successfully creating a stack

Stack stack1 = os.heat().stacks().create(Builders.stack().name("teststack").template(hotContent).timeoutMins(5L).build());

the stack1 object's state consists only of the 'id' field value and all other are set to null. To view the details of the Stack I just created I must perform the following:

Stack stack2 = os.heat().stacks().getDetails("teststack", stack.getId());

then stack2 has all the fields populated with the correct values. Change is the status is not reflected by the original stack1 object.
